docs/Adding-Software-to-Homebrew.md

```diff
@@ -8,46 +8,139 @@ Before you start, please check the open pull requests for [Homebrew/homebrew-cor
 
-1. You will now have to develop the boilerplate code from `brew create` into a full-fledged formula. Your main references will be the [Formula Cookbook](Formula-Cookbook.md), similar formulae in Homebrew, and the upstream documentation for your chosen software. Be sure to also take note of the Homebrew documentation for writing [Python](Python-for-Formula-Authors.md) and [Node](Node-for-Formula-Authors.md) formulae, if applicable.
+1. If you're starting from scratch, you can use the [`brew create` command](Manpage.md#create-options-url) to produce a basic version of your formula. This command accepts a number of options and you may be able to save yourself some work by using an appropriate template option like `--python` / `--go`.
 
-1. Make sure you write a good test as part of your formula. Refer to the [Add a test to the formula](Formula-Cookbook.md#add-a-test-to-the-formula) section of the Cookbook for help with this.
+1. You will now have to develop the boilerplate code from `brew create` into a complete formula. Your main references will be the [Formula Cookbook](Formula-Cookbook.md), similar existing formulae, and the official documentation of the software you're packaging. Be sure to also take note of the Homebrew documentation for writing [Python](Python-for-Formula-Authors.md) and [Node](Node-for-Formula-Authors.md) formulae, if applicable.
 
-1. Try installing your formula using `brew install --build-from-source <formula>`, where *\<formula>* is the name of your formula. If any errors occur, correct your formula and attempt to install it again. The formula installation should finish without errors by the end of this step.
+1. Make sure you write a good test as part of your formula. Refer to the [Add a test to the formula](Formula-Cookbook.md#add-a-test-to-the-formula) section of the Formula Cookbook for help with this.
 
 If you're stuck, ask for help on GitHub or the [Homebrew discussion forum](https://github.com/orgs/Homebrew/discussions). The maintainers are very happy to help but we also like to see that you've put effort into trying to find a solution first.
 
 ### Testing and auditing the formula
 
-1. Run `brew audit --strict --new-formula --online <formula>` with your formula. If any errors occur, correct your formula and run the audit again. The audit should finish without any errors by the end of this step.
+1. Test your formula installation using `brew install --formula --build-from-source $(brew --repository homebrew/core)/Formula/<prefix>/<formula>.rb`, where `<prefix>` is the first letter of your formula's name and `<formula>` is the name of your formula. Correct any errors that occur and repeat the installation until it completes without errors.
+
+1. Additionally, verify the proper uninstallation of the formula by using the command `brew uninstall --formula <formula>`, ensuring it removes the software without issues.
```

Comment:
I don't think this is necessary; there's no real reason to think it won't uninstall properly the vast majority of the time.

Reply:
I pretty much only added this as it seems to be a requirement for Casks (at least based on the pull request template, can't remember if it was said elsewhere as well); and was trying to align the steps for both.

If it's not needed for either, awesome, less to do. If it's specifically not needed for Formulae but for some reason is for Casks then can have them different.

Reply:
Sounds like Casks only:

> This is relevant to casks because uninstall logic is often non-trivial there.
> 
> _Originally posted by @MikeMcQuaid in https://github.com/Homebrew/brew/pull/16327#discussion_r1425270148_

Can remove it from this section.

Reply:
Yeh, agreed, just need for casks 👍🏻

Reply:
To keep things 'consistent' between the Formula and Casks sections, instead of removing this entirely, I want to add some words to explicitly say something akin to:

> Unlike when making a Cask, you don't need to explicitly test uninstalling for Formula

That way they are more 'consistently aligned'; and someone familiar with one process can see how/why it differs, rather than guessing/wondering if the docs are outdated/don't mention it by accident/etc.

Reply:
@MikeMcQuaid Fixed in 5c83d3f04132e785af7682b604a75969f0fc1f49

I'll leave this comment thread open for visibility for now; but once you check it and are happy, leave a comment and then feel free to mark it resolved.